### DISCUSSION TITLE
How to deploy custom implementation using App Service and OpenAI Service on Azure

### DISCUSSION THREAD

**Discussion by jeannot-prvlimburg** [microsoft/sample-app-aoai-chatGPT#498] [Q&A]
Hi all, I recently deployed a webapp (App Service) using a custom LLM (OpenAI Service). I have now forked and customized this repository to change the front-end of the application. However, when I replace this repository in the App Service with my own custom version of this repository, nothing changes. Do you have any idea how to fix this issue?

**Comment by jeannot-prvlimburg** [comment]
Ensure updates are made to the branding properties, encompassing logos and titles, specifically within `index-87c2be60.js`, despite their potential redundancy in other files.

**Reply by cytrus2012** [reply]
Hi, could anyone please help out. Any changes that I try to make to the app does not reflect on the frontend. Am I doing something wrong? Thanks

**Reply by jeannot-prvlimburg** [reply]
What modifications did you make? Where were these changes implemented? Have you committed your changes to the appropriate branch? Does your Azure application point to the correct repository and branch?

**Reply by cytrus2012** [reply]
we have tried in 2 different way, first was trying to change files(index.html) from the kudu side and restarting, which did not work. 
We tried every way to deploy the app even we created our own repository on our local machine copy everything that we had and pushed it into our remote repository. It's like we created the app and pushed into a github in a fresh repo. 

Still does not work.. We also tried to fork it. But it also did not work at this point, we are stucked.
We wanted to make simple changes with the name something else instead of contoso and the image.

We used appropriate branch regarding what azure indicates. it was an ftp deployment, manually so yes it was also pointing correctly.

**Reply by iMicknl** [reply]
See https://github.com/microsoft/sample-app-aoai-chatGPT?tab=readme-ov-file#updating-the-default-chat-logo-and-headers. After you have changed the source material, you will need to run `start.sh` or `start.cmd`. See https://github.com/microsoft/sample-app-aoai-chatGPT?tab=readme-ov-file#deploy-with-the-azure-cli.

The following steps work for me:
1. Fork the repository on GitHub and set it to private
2. Make the (frontend) changes
3. Run `start.sh` or `start.cmd` to regenerate all front-end assets. Commit all your changes and push your changes.
4. In Azure, navigate to your Web Application and select Deployment Center -> Settings.
5. Disconnect the source, and configure the source to your private repository on GitHub. 
6. Press sync.

**Comment by gochristoph** [comment]
Just to add, if anyone does not require to run the app locally but just wants to make a small change, you can go for a more granular approach that does not require installing all dependencies: 

0. Execute only once:
  0.1 Install Node.js (e.g., `brew install node` on macOS) if that is not already the case.
  0.2 Install Node.js dependencies (`cd frontend; npm install`).
1. Execute after every change to the frontend:
1.1 `npm run build`.
1.2 Commit and push the files that were created by the build (index-*.js and index-*.js.map in static/assets/) in addition to your changes to the frontend source files.

For reference:
* [start.sh](https://github.com/microsoft/sample-app-aoai-chatGPT/blob/main/start.sh) / [start.cmd](https://github.com/microsoft/sample-app-aoai-chatGPT/blob/main/start.cmd) 
* Corresponding part in the [documentation](https://github.com/microsoft/sample-app-aoai-chatGPT/blob/main/README_azd.md).
